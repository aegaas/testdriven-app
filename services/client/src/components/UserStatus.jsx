import React, { Component } from "react";
import axios from "axios";
import { Link } from "react-router-dom";

class UserStatus extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: "",
      id: "",
      username: ""
    };
  }
  componentDidMount() {
    if (this.props.isAuthenticated) {
      this.getUserStatus();
    }
  }
  getUserStatus(event) {
    const options = {
      url: `${process.env.REACT_APP_USERS_SERVICE_URL}/auth/status`,
      method: "get",
      headers: {
        "Content-Type": "application/json",
        Authorization: `Bearer ${window.localStorage.authToken}`
      }
    };
    return axios(options)
      .then(res => {
        this.setState({
          email: res.data.data.email,
          id: res.data.data.id,
          username: res.data.data.username
        });
      })
      .catch(error => {
        console.log(error);
      });
  }
  render() {
    if (!this.props.isAuthenticated) {
      return (
        <p>You must be logged in to view this. Click <Link to="/">here</Link> to log back in.</p>
      )
    }
    return (
      <div>
        <ul>
          <li>
            <strong>ID:</strong>
            <span className="value">{this.state.id}</span>
          </li>
          <li>
            <strong>Username:</strong>
            <span className="value">{this.state.username}</span>
          </li>
          <li>
            <strong>Email:</strong>
            <span className="value">{this.state.email}</span>
          </li>
        </ul>
      </div>
    );
  }
}

export default UserStatus;
