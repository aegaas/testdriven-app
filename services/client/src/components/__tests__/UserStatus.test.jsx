import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

import UserStatus from '../UserStatus';


beforeAll(() => {
  global.localStorage = {
    getItem: () => 'someToken'
  };
});

let mock = new MockAdapter(axios);
let options = {
  headers: {
    Authorization: "Bearer someToken"
  }
}
mock.onGet(`${process.env.REACT_APP_USERS_SERVICE_URL}/auth/status`, options).reply(200, {
  data: {
    id: 1,
    username: 'aegaas',
    email: 'me@aaronegaas.com'
  }
});

xdescribe('Not Authenticated', () => {
  const testData = {
    isAuthenticated: false
  }
  it('UserStatus renders a message to login', () => {
    const wrapper = shallow(<UserStatus isAuthenticated={testData.isAuthenticated} />);
    const element = wrapper.find('p');
    expect(element.length).toBe(1);
    expect(element.get(0).props.children[0]).toContain('You must be logged in to view this.');
  });

  it('UserStatus renders a snapshot properly', () => {
    const tree = renderer.create(
      <Router><UserStatus isAuthenticated={false} /></Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

xdescribe('Authenticated', () => {
  const testData = {
    isAuthenticated: true
  }
  it('UserStatus renders a message to login', () => {
    const wrapper = shallow(<UserStatus isAuthenticated={testData.isAuthenticated} />);
    const element = wrapper.find('li span.value');
    expect(element.length).toBe(3);
    console.log(element.get(0).props.text)
    expect(element.get(0).props.children[0]).toContain('1');
    expect(element.get(1).props.children[0]).toContain('aegaas');
    expect(element.get(2).props.children[0]).toContain('me@aaronegaas.com');
  });

  it('UserStatus renders a snapshot properly', () => {
    const tree = renderer.create(
      <Router><UserStatus isAuthenticated={false} /></Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
