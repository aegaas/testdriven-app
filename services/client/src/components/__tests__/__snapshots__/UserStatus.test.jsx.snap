// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authenticated UserStatus renders a snapshot properly 1`] = `
<p>
  You must be logged in to view this. Click 
  <a
    href="/"
    onClick={[Function]}
  >
    here
  </a>
   to log back in.
</p>
`;

exports[`Not Authenticated UserStatus renders a snapshot properly 1`] = `
<p>
  You must be logged in to view this. Click 
  <a
    href="/"
    onClick={[Function]}
  >
    here
  </a>
   to log back in.
</p>
`;
